overlord/snapstate/snapmgr.go

```diff
@@ -562,16 +559,22 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	// No need to undo "snaps" in state here. The only chance of
-	// having the new state there is a working doLinkSnap call.
-
 	// relinking of the old snap is done in the undo of unlink-current-snap
 
-	newInfo, err := retrieveInfo(ss.Name, snapst.Candidate)
+	n := len(snapst.Sequence)
+	cand := snapst.Sequence[n-1]
```

Comment:
I wouldn't worry about the distinctino of n == 1 and otherwise below. snapst is getting out of scope entirely in a few lines. So we might do it all just with:
```
snapst.Candidate = snapst.Sequence[len(snapst.Sequence)-1]
snapst.Sequence = snapst.Sequence[:len(snapst.Sequence)-1]
```